integration_tests/run_pyspark_from_build.sh

```diff
@@ -129,9 +134,26 @@ else
     export PYSP_TEST_spark_ui_showConsoleProgress='false'
     export PYSP_TEST_spark_sql_session_timeZone='UTC'
     export PYSP_TEST_spark_sql_shuffle_partitions='12'
-    # prevent cluster shape to change - and fail quicker rather than retry
-    export PYSP_TEST_spark_task_maxFailures='1'
+    # prevent cluster shape to change
     export PYSP_TEST_spark_dynamicAllocation_enabled='false'
+
+    # Set spark.task.maxFailures for most schedulers.
+    #
+    # Local (non-cluster) mode is the exception and does not work with `spark.task.maxFailures`.
+    # It requires two arguments to the master specification "local[N, K]" where
+    # N is the number of threads, and K is the maxFailures (otherwise this is hardcoded to 1,
+    # see https://issues.apache.org/jira/browse/SPARK-2083).
+    export PYSP_TEST_spark_task_maxFailures="$SPARK_TASK_MAXFAILURES"
+
+    if ((NUM_LOCAL_EXECS > 0)); then
+      export PYSP_TEST_spark_master="local-cluster[$NUM_LOCAL_EXECS,$CORES_PER_EXEC,$MB_PER_EXEC]"
```

Comment:
There is no requirement that this script be used in local mode!!!. It started out that way, but is now used everywhere, and we probably should rename it.  We cannot assume this.

Reply:
Sorry I am confused. `NUM_LOCAL_EXECS > 0` is unrelated to this change, and if this variable is not specified it defaults to 0. Below, I am trying (perhaps failing) to check that the master was set.

Reply:
Sorry this script has changed so much since I wrote it and now it is very confusing.

Reply:
@revans2 The default is 0 and does not interfere with the decision to use or not to use local-cluster mode, but if the user sets it to a positive value it's exactly what the user is signalling to the script https://github.com/NVIDIA/spark-rapids/blob/8d3efb8ffba02da69768f7d69bc148da3a01afa9/integration_tests/README.md#spark-execution-mode